pattern/junkyard-styled-innersourcing

nyeates opened the pull request:
People don't share something so others can use it; they only share it for the sake of sharing. They wouldn't mind if no one uses it (a bit of a graveyard: "InnerSource is where modules go to die")

Commit:
Created

This is also in master, and it really shouldn't be, as its not a mature pattern at all. Starting a PR to start a proper Review Process.

rshanmer commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,21 @@
+
+* SW component not designed for reuse because the schedule was too tight and did not allow for the extra effort needed
+* Did not have the knowledge or training to create components that can b reused.
+
```
See other suggestion to pick one context.  [test]

NewMexicoKid committed:
Update junkyard-styled-innersourcing.md

Merging in comments to this point except for the one to split this into two patterns based on the two situations listed in the context.

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+## Context
+Two situations:
+
+* SW component not designed for reuse because the schedule was too tight and did not allow for the extra effort needed
```
I would argue that the effort for designing SW for reuse should only be spent, if it is crystal clear that it in fact will be reused. Otherwise, YAGNI applies. So either it was clear that it would be reused and it wasn't designed that way for the reasons you mentioned, or it wasn't clear and therefore not designed for reuse (and rightly so).

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+* Did not have the knowledge or training to create components that can b reused.
+
+## Problem
+* How do you encourage people to share whatever code they write?
```
Hmm, the pattern title suggests to me that people are sharing, just not stuff that is ready to be reused. How about we rephrase the problem statement to s.th. like _Shared code is not reused because it was not written with reuse in mind or is not reusable._

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+* A developer might be a person who likes to share their code; one that likes others to read it (and maybe comment on it. But mostly they like to write code. "Hey, I wrote this nifty gzorp and you might like it (but its no skin off my nose if you don't)"
+* the code is crap
+* the code doesn't address any of the company's problems (at the moment anyway)
+* How does the person let others know that the code exists?
```
I was wondering if this is a force ... as an idea: _The less effort is spent on publicising shared code the less likely it is that it will be found and reused._

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+## Problem
+* How do you encourage people to share whatever code they write?
+
+## Forces
```
@rshanmer : Some of the forces in this pattern are formulated as questions, rather than e. g. tradeoffs. Is this a valid way to phrase forces?

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+* How do we politely take their freely offered code and harden it or otherwise prepare it for production without having the original contributor feel like we are saying their code is crap?
+
+## Solution
+* Support all contributions to the shared repository.
```
I like the idea behind this but I feel this needs to be fleshed out a little. Here are some ideas:
- Support the users in writing proper `readme.md` and `contribute.md` files, e. g. by providing templates
- Announce new shared repositories in a weekly/monthly blog post and give the authors a chance to make themselves and their SW known to others

gruetter commented on junkyard-styled-innersourcing.md:
```diff
@@ -0,0 +1,35 @@
+* How does the person let others know that the code exists?
+* We want the person to contribute code, so what roadblocks need to be removed so that they continue making their code available?
+* How do we politely take their freely offered code and harden it or otherwise prepare it for production without having the original contributor feel like we are saying their code is crap?
+
```
Suggestion for additional force: _The more responsive the original author is to requests for changes coming from potential re-users of her code, the more likely it is that it will be reused._
